### PR TITLE
Jl form functionality cleanup

### PR DESCRIPTION
**Pull request by jleonardw9**

### CODE REVIEW

**Review by wtholt — APPROVED** [review]

File: app/helpers/additional_details/application_helper.rb [hunk 1]
```diff
@@ -1,2 +1,9 @@
 module AdditionalDetails::ApplicationHelper
+
+  def convert_time(time)
+    modifier = ''
+    ((time.first == '0') || (time[0..1] == '10') || (time[0..1] == '11')) ? modifier = ' AM' : modifier = ' PM'
+
+    DateTime.strptime(time.gsub(':', ''), '%H%M').strftime('%l:%M') + modifier
```
Comment (kayla-glick) [inline app/helpers/additional_details/application_helper.rb]: There should be a placeholder for AM/PM in DateTime.strptime. %p should do it

File: app/views/additional_details/submissions/_submission_info.html.haml [hunk 2]
```diff
@@ -9,6 +9,10 @@
             %p.list-group-item-text
               - if item.item_type == 'multiple_dropdown'
                 = qr.content.delete('[]').delete('\"')
```
Comment (kyle-glick) [inline app/views/additional_details/submissions/_submission_info.html.haml]: Can you add the gsub here as well?

**Review by kayla-glick — CHANGES_REQUESTED** [review]

File: app/helpers/additional_details/application_helper.rb (hunk 1 above)
Comment (jleonardw9) [reply app/helpers/additional_details/application_helper.rb]: I kind of like my modifier code, so I think I'm going to leave it in.

File: app/views/additional_details/submissions/_submission_info.html.haml (hunk 2 above)
Comment (jleonardw9) [inline app/views/additional_details/submissions/_submission_info.html.haml]: That's Will's code and it's doing it's job, so I'm not going to mess with it.

**Review by jwiel86 — APPROVED** [review]

File: app/helpers/additional_details/application_helper.rb (hunk 1 above)
Comment (kayla-glick) [reply app/helpers/additional_details/application_helper.rb]: Can you explain what the modifier is doing? I'm a bit confused about the '0' || '10' || '11' part

Comment (jleonardw9) [reply app/helpers/additional_details/application_helper.rb]: It's converting military time to regular time. So if a time string has either '0', '10', or '11' as the leading characters then it is in the morning. All else are PM.

**Comment by bmic-development** [comment]
Time format is something like 02:01:01 or 11:21:23 or 23:13:13.  So it’s looking at if it start with 0 or first 2 numbers are 10,11 then it’s AM otherwise it’s PM.

File: app/helpers/additional_details/application_helper.rb (hunk 1 above)
Comment (kayla-glick) [reply app/helpers/additional_details/application_helper.rb]: You could use `DateTime.strptime(time, '%H:%M').strftime('%l:%M %p')` to do the same thing all in 1 line

**Review by kayla-glick — CHANGES_REQUESTED** [review]

File: app/views/additional_details/submissions/_submission_info.html.haml (hunk 2 above)
Comment (kyle-glick) [inline app/views/additional_details/submissions/_submission_info.html.haml]: It looks at though it's performing the same function as the checkbox gsub. @wtholt is that correct? If so, could we combine the checkbox and multiple_dropdown logic into one if statement
`if ['multiple_dropdown', 'checkbox'].include?(item.item_type)`

**Comment by amcates** [comment]
@jleonardw9 I have to agree with Kyle.  Using the strptime method instead of custom code would be better in this case.

File: app/helpers/additional_details/application_helper.rb (hunk 1 above)
Comment (jleonardw9) [reply app/helpers/additional_details/application_helper.rb]: Totally missed this comment last week. Your suggestion worked great, should be up there now.

**Comment by jleonardw9** [comment]
Yeah, I agree as well. Just changed it.

File: app/views/additional_details/submissions/_submission_info.html.haml [hunk 3]
```diff
@@ -9,6 +9,10 @@
...
+              - elsif item.item_type == 'time'
+                = convert_time(qr.content)
+              - elsif item.item_type == 'checkbox'
+                = qr.content.gsub(/["\[\]]/, '')
```
Comment (kayla-glick) [inline app/views/additional_details/submissions/_submission_info.html.haml:15]: I tested `qr.content.delete('[]').delete('\"')` and `qr.content.gsub(/["\[\]]/, '')` and they appear to be doing the same thing: Replacing/removing the characters []". We should combine this block and the if block into a single statement. 
`if ['checkbox', 'multiple_dropdown'].include?(item.item_type)`
`  qr.content.delete('[]\"')` or `qr.content.gsub(/["\[\]]/, '')`

**Review by kayla-glick — DISMISSED** [review]

File: app/views/additional_details/submissions/_submission_info.html.haml (hunk 3 above)
Comment (amcates) [reply app/views/additional_details/submissions/_submission_info.html.haml:15]: I would label that a refactor and make another pull request.  Not necessary to proceed.  In the same thought you could change 'qr.content.gsub' to another method call similar to convert_time.

Comment (kayla-glick) [reply app/views/additional_details/submissions/_submission_info.html.haml:15]: Fair enough.

**Review by Stuart-Johnson — APPROVED** [review]

**Pull request merged**